### ISSUE TITLE
Add group name element

### ISSUE THREAD

**Issue by markn86** [markn86/moodle-mod_customcert#155]
This would be useful - see https://moodle.org/mod/forum/discuss.php?d=363376.

**Comment by Nata27** [comment]
Hi Marc,
will it also be possible to add the group name to the email, which is sent upon issuing the certificate? This would be very important for us. When are you planning on implementing this feature?
Thanks in advance.

**Comment by markn86** [comment]
Hi @Nata27,

I still haven't added the group element to the custom certificate yet. I need to do that before I add it to the email (which should be a separate issue).

I work on this in my spare time so am not sure when I can get to it. There is already a lot of other popular requests. I have a week coming up where I will spend most of it working on this plugin, but I am not sure I will get to this issue.

However, if you are able to provide a patch I would be happy to review it.

**Comment by Nata27** [comment]
Hi Mark,
thanks for your reply. Unfortunately I'm not much into programming moodle plug-ins myself, so I guess we'll just have to find a workaround until you get to it.

**Comment by johnpap21** [comment]
I would agree, it is something we would like to have too on our certificates.

**Comment by dheuring** [comment]
I've had to do a work around to get the group on the certificate by creating a separate user field with the group listed there, but it is not ideal at all and requires additional steps to get it to show on the certificate.  Without pressuring anyone, is there likelihood this feature can be added in the near future?

**Comment by markn86** [comment]
I have a week coming up shortly after the 3.7 release where I will get a chance to work on these feature requests.

**Comment by hcmendes** [comment]
Hi,
Aside from the group name, is it possible to add the group's teacher name to the certificate? It would be very helpful

**Comment by mdjnelson** [comment]
That would be a totally different issue and element to work on.

**Comment by koenraad26** [comment]
Hello mdjnelson,

I've tried to create the groupName element.

Can you help me with the query?

 
[groupname.zip](https://github.com/mdjnelson/moodle-mod_customcert/files/10782584/groupname.zip)

**Comment by koenraad26** [comment]
[groupname.zip](https://github.com/mdjnelson/moodle-mod_customcert/files/10783958/groupname.zip)

I've created the groupname element.

Sorry i've to follow a course how to use github.

**Comment by koenraad26** [comment]
[groupname.zip](https://github.com/mdjnelson/moodle-mod_customcert/files/11020720/groupname.zip)

**Comment by mdjnelson** [comment]
Are you able to turn this into a pull request? Thanks.

**Comment by mdjnelson** [comment]
I have created https://github.com/mdjnelson/moodle-mod_customcert/pull/831. Will submit to `main` once i'm happy with it then backport to 5.2. I am trying to stick to only keeping new features in the `main` branch but in this case since 5.3 won't be out for a while and this has been requested quite a few times - so ill backport to 5.2.

**Comment by mdjnelson** [comment]
Done. Will be in the 5.2.3 release.
